src/HospitalLibrary/Rooms/RoomEquipmentService.cs

```diff
@@ -0,0 +1,107 @@
+﻿using HospitalLibrary.Rooms.Dtos;
+using HospitalLibrary.Rooms.Interfaces;
+using HospitalLibrary.Rooms.Model;
+using HospitalLibrary.Shared.Interfaces;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace HospitalLibrary.Rooms
+{
+    public class RoomEquipmentService: IRoomEquipmentService
+    {
+        private IUnitOfWork _unitOfWork;
+        private IRoomEquipmentValidator  _roomEquipmentValidator;
+
+        public RoomEquipmentService(IUnitOfWork unitOfWork, IRoomEquipmentValidator roomEquipmentValidator)
+        {
+            _unitOfWork = unitOfWork;
+            _roomEquipmentValidator = roomEquipmentValidator;
+        }
+
+        public Task<IEnumerable<RoomEquipment>> searchEquipmentInRoom(RoomEquipmentDTO roomEquipmentDTO)
+        {
+            switch (checkSearchInputRoom(roomEquipmentDTO))
+            {
+                case 0: 
+                    return  _unitOfWork.RoomEquipmentRepository.GetAllByCombineSearchInRoom(roomEquipmentDTO);
+                case 1:
+                    return _unitOfWork.RoomEquipmentRepository.GetAllByNameSearchInRoom(roomEquipmentDTO);
```

Comment:
This is ok, but in a longterm can be a problem. If you add more cases, the method can get really big. Think about some kind of filter pipeline which you can easily apply in repositories.